pkg/operator/starter.go

```diff
@@ -64,12 +67,39 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 
 	logLevelController := loglevel.NewClusterOperatorLoggingController(operatorClient, controllerContext.EventRecorder)
 
+	// As this operator does not manage any component/workload, report this operator as available and not progressing by default.
+	// TODO: Revisit this with full controller at some point.
+	operatorController := factory.New().ResyncEvery(10*time.Second).WithSync(func(ctx context.Context, controllerContext factory.SyncContext) error {
```

Comment:
@deads2k i think we missing this bit in library-go... if you have generic operator (not static pod, not workloads), nothing really sets the `FooAvailable` anywhere, which means the status controller will union there as `"Unknown"` with `NoData` as reason... The CVO then freak out, as there is nothing that moves the operator Progressing to false and Available to true (it will stay "Unknown"). I'm 99% sure that is the reason why the e2e tests are failing.

Since this operator does not have any workload to manage and does not have any controller that would set "FooAvailable", I think it is reasonable to have something like below that "default" the operator conditions on startup (which is why I placed this to starter). WDYT?

Reply:
I prefer this as a separate controller in a separate package or you'll have people adding their, "just a small thing" here.